3p/ampcontext.js

```diff
@@ -171,8 +171,10 @@ export class AmpContext {
     // to check the name attribute as it has been bypassed.
     // TODO(alanorozco): why the heck could AMP_CONTEXT_DATA be two different
     // types? FIX THIS.
-    if (!this.win_.AMP_CONTEXT_DATA) {
+    if (!this.win_.AMP_CONTEXT_DATA && !this.win_.sf_) {
```

Comment:
Feels like this could be cleaned up a bit, I suggest the following.

if (this.win_.sf_ && this.win_.sf_.cfg) {
   this.setupMetadata_(this.win.sf_.cfg);
} else if (this.win_.AMP_CONTEXT_DATA) {
   if (typeof this.win_.AMP_CONTEXT_DATA== 'string') {
      this.sentinel = this.win_.AMP_CONTEXT_DATA;
   } else {
      this.setupMetadata_(this.win_.AMP_CONTEXT_DATA);
   }
}

Is there an isString utility function you can use instead of typeof?
Do you need to verify this.win_.sf_.cfg is an object?